test/src/e2e_vm_tests/test_programs/should_fail/storage_types/src/main.sw

```diff
@@ -14,7 +14,12 @@ storage {
     map1: StorageMap<u32, u32> = StorageMap{},
 }
 
-impl Contract {
+abi MyContract {
+    #[storage(read, write)]
+    fn main() -> u64;
+}
+
+impl MyContract for Contract {
```

Comment:
this should be an `impl Contract` and not an `impl abi for Contract`, as a negative test, correct?

Reply:
I thought this test should be fixed because it tests an orthogonal feature (storage types). The negative test for this fix is in `self_impl_contract`: https://github.com/FuelLabs/sway/blob/19471c7942ad1dd1cb4ecb45928808f16ae01bbe/test/src/e2e_vm_tests/test_programs/should_fail/self_impl_contract/src/main.sw